src/planner/ostree.rs

```diff
@@ -0,0 +1,339 @@
+use crate::{
+    action::{
+        base::{CreateDirectory, CreateFile, RemoveDirectory},
+        common::{ConfigureInitService, ConfigureNix, CreateUsersAndGroups, ProvisionNix},
+        linux::{ProvisionSelinux, StartSystemdUnit, SystemctlDaemonReload},
+        StatefulAction,
+    },
+    error::HasExpectedErrors,
+    planner::{Planner, PlannerError},
+    settings::CommonSettings,
+    settings::{InitSystem, InstallSettingsError},
+    Action, BuiltinPlanner,
+};
+use std::{collections::HashMap, path::PathBuf};
+
+use super::{
+    linux::{
+        check_nix_not_already_installed, check_not_nixos, check_not_wsl1, check_systemd_active,
+        detect_selinux,
+    },
+    ShellProfileLocations,
+};
+
+/// A planner for Linux installs
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[cfg_attr(feature = "cli", derive(clap::Parser))]
+pub struct Ostree {
+    /// Where `/nix` will be bind mounted to.
+    #[cfg_attr(feature = "cli", clap(long, default_value = "/var/home/nix"))]
+    persistence: PathBuf,
```

Comment:
I've added this flag because the steam deck has it too.
I wonder though if it's really necessary to make this configurable.

Reply:
The steam deck planner has removed this in the latest version since the SteamOS team added a `/home/.steamos/offload/nix` folder.

I think it's a good idea to keep this configurable since users have previously expressed a desire to be able to change the location on their steam deck, so I imagine it will be the case here too.